src/raylib/model/Model.hpp

```diff
@@ -124,9 +146,15 @@ namespace raylib
             /// @param mapType the map type to replace the texture
             void setMaterialMapTexture(const raylib::textures::Texture2D &texture, int materialId, int mapType);
 
+            /// Set the model as owner of the mesh or not.
+            /// @note Model owning a mesh will unload it on destruction.
+            ///
+            /// @param ownsMesh If the model owns the mesh.
+            void setOwnsMesh(bool ownsMesh = true);
```

Comment:
This is very sketchy at best. If you want to shared something between different objects, please use a shared_ptr instead.

Reply:
I know but shared pointers requires to be allocated on the heap whereas this solution doesn't.